go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/Masterminds/semver/v3 v3.1.0
 	github.com/Shopify/sarama v1.27.2
 	github.com/blevesearch/bleve v1.0.13
-	github.com/dgraph-io/badger/v4 v4.1.0
+	github.com/dgraph-io/badger/v4 v4.0.2-0.20230731151745-c2e611ac7cc6
```

Comment:
why are we downgrading from `4.1.0` to `4.0.2`?

Reply:
Because the v4.1.0 is on release/v4 branch and v4.0.2 is on main branch. It looks that way.

Reply:
Not sure I understand it, I will sync up offline.

Reply:
@mangalaman93 is correct ^ this is an artifact of the git branch structure in the badger repo.  Go automatically generates these version numbers based the git branches.